src/ui/common/src/components/pages/metric/id/index.tsx

```diff
@@ -50,109 +42,68 @@ const MetricDetailsPage: React.FC<MetricDetailsPageProps> = ({
+    // Load workflow dag result if it's not cached
+    if (
+      !workflowDagResultWithLoadingStatus ||
+      isInitial(workflowDagResultWithLoadingStatus.status)
+    ) {
       dispatch(
-        handleGetOperatorResults({
+        handleGetWorkflowDagResult({
           apiKey: user.apiKey,
+          workflowId,
           workflowDagResultId,
-          operatorId: metricOperatorId,
         })
       );
     }
-  }, [workflow.dagResults]);
+  }, []);
 
   useEffect(() => {
-    if (!metricResult || !metricResult.result) {
-      return;
+    // Load artifact history once workflow dag results finished loading
+    // and the result is not cached
+    if (
+      !artifactHistoryWithLoadingStatus &&
+      !artifactId &&
+      !!workflowDagResultWithLoadingStatus &&
+      !isInitial(workflowDagResultWithLoadingStatus.status) &&
+      !isLoading(workflowDagResultWithLoadingStatus.status)
+    ) {
+      dispatch(
+        handleListArtifactResults({
+          apiKey: user.apiKey,
+          workflowId,
+          artifactId,
+        })
+      );
     }
+  }, [workflowDagResultWithLoadingStatus, artifactId]);
 
-    document.title = `${metricResult.result.name} | Aqueduct`;
-  });
-
-  // Set up different metric input types for rendering in the inputs list.
-  // TODO: transform/handle response from API and render these appropriately.
-  const mockMetricInputs = [
-    {
-      name: 'bool_artifact',
-      nodeType: NodeType.BoolArtifact,
-      nodeIcon: boolArtifactNodeIcon,
-    },
-    {
-      name: 'checkOperator',
-      nodeType: NodeType.CheckOp,
-      nodeIcon: checkOperatorNodeIcon,
-    },
-    {
-      name: 'database_node',
-      nodeType: NodeType.LoadOp,
-      nodeIcon: databaseNodeIcon,
-    },
-    {
-      name: 'function_operator_node',
-      nodeType: NodeType.FunctionOp,
-      nodeIcon: functionOperatorNodeIcon,
-    },
-    {
-      name: 'generic_artifact_node',
-      nodeType: NodeType.GenericArtifact,
-      nodeIcon: genericArtifactNodeIcon,
-    },
-    {
-      name: 'image_artifact_node',
-      nodeType: NodeType.ImageArtifact,
-      nodeIcon: imageArtifactNodeIcon,
-    },
-    {
-      name: 'jsonArtifactNode',
-      nodeType: NodeType.JsonArtifact,
-      nodeIcon: jsonArtifactNodeIcon,
-    },
-    {
-      name: 'metric_operator_node',
-      nodeType: NodeType.MetricOp,
-      nodeIcon: metricOperatorNodeIcon,
-    },
-    {
-      name: 'numeric_artifact_node',
-      nodeType: NodeType.NumericArtifact,
-      nodeIcon: numericArtifactNodeIcon,
-    },
-    {
-      name: 'string_artifact_node',
-      nodeType: NodeType.StringArtifact,
-      nodeIcon: stringArtifactNodeIcon,
-    },
-    {
-      name: 'table_artifact_node_icon',
-      nodeType: NodeType.TableArtifact,
-      nodeIcon: tableArtifactNodeIcon,
-    },
-  ];
+  useEffect(() => {
+    if (!!operator) {
+      document.title = `${operator.name} | Aqueduct`;
+    }
+  }, [operator]);
 
   const artifactTypeToIconMapping = {
```

Comment:
I think it would be helpful if you could move this out to a common file so other details page can use this mapping.